packages/lit-dev-content/src/components/litdev-search.ts

```diff
@@ -208,81 +342,145 @@ class LitDevSearch extends LitElement {
   }
 
   /**
    * Load the search index.
    */
   firstUpdated() {
-    // Late load search index when page is idle.
-    if ('requestIdleCallback' in window) {
-      (window as any).requestIdleCallback(() => this.loadSearchIndex());
-    } else {
-      setTimeout(() => this.loadSearchIndex(), 500);
-    }
+    this._loadSearchIndex();
+    // required for popping up on hydration
+    this._isFocused = !!this.shadowRoot?.activeElement;
+  }
+
+  updated(changed: PropertyValues) {
+    super.updated(changed);
+
+    if (changed.has('_isFocused') || changed.has('_suggestions')) {
+      const isExpanded = this._isFocused && this._suggestions.length > 0;
+      if (!isExpanded) {
+        return;
+      }
 
-    this.input.placeholder = 'Search';
+      this._positionPopup();
+    }
   }
 
   /**
    * Repopulate suggestions with each input event.
    */
-  private handleInput(e: InputEvent) {
-    this.searchText = (e.target as HTMLInputElement).value ?? '';
-    this.querySearch(this.searchText);
+  private _onInput(e: InputEvent) {
+    this._searchText = (e.target as HTMLInputElement).value ?? '';
+    this._querySearch(this._searchText);
   }
 
   /**
    * Populate suggestion dropdown from query.
    *
    * An empty query clears suggestions.
    */
-  private querySearch(query: string) {
+  private _querySearch(query: string) {
     const trimmedQuery = query.trim();
+    this._selectedIndex = -1;
     if (
-      !LitDevSearch.siteSearchIndex ||
+      !LitDevSearch._siteSearchIndex ||
       trimmedQuery === '' ||
       trimmedQuery.length < 2
     ) {
-      this.suggestions = [];
+      this._suggestions = [];
       return;
     }
-    this.suggestions = (
-      LitDevSearch.siteSearchIndex.search(trimmedQuery) ?? []
+    this._suggestions = (
+      LitDevSearch._siteSearchIndex.search(trimmedQuery) ?? []
     ).slice(0, 10) as unknown as Suggestion[];
+    this._selectedIndex = -1;
   }
 
   /**
    * Handle key press with side effects.
    *  - "Enter" finds the selected option and navigates to the page.
    */
-  private handleKeyDown(e: KeyboardEvent) {
-    if (this.searchOptionElements.length > 0 && e.key === 'Enter') {
-      // Navigate to checked element.
-      for (const el of this.searchOptionElements) {
-        if (el.checked as boolean) {
-          this.navigate(el.relativeUrl);
-          return;
+  private _onKeydown(e: KeyboardEvent) {
+    switch (e.key) {
+      case 'ArrowDown':
+        this._selectNext();
+        break;
+      case 'ArrowUp':
+        this._selectPrevious();
+        break;
+      case 'Enter':
+        this._select();
+        break;
+      case 'Escape':
+        const oldText = this._searchText;
+        this._searchText = '';
+        this._suggestions = [];
+        this._selectedIndex = -1;
+        // prevent the input from closing dialog if there was text
+        if (oldText.trim()) {
+          e.stopPropagation();
         }
+        break;
```

Comment:
I'm slightly unsure what `oldText` is being used for here.

Currently when manually testing, pressing escape always closes and clears the input. Previously pressing escape while focused on the suggestions would only close them and not clear input. And pressing escape while focussed on input would clear input.

Reply:
The comment should be updated to say "drawer" instead. This is if you put it in a drawer, you're typing and you hit escape, it will clear it but it won't close the dialog.

If the searchbox is empty and you hit escape, you likely wanted to close the drawer